server/src/main/java/org/apache/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -258,9 +258,73 @@ public int markSegmentsAsUnusedWithinInterval(String dataSource, Interval interv
 
   /**
    * Fetches all the pending segments, whose interval overlaps with the given
-   * search interval from the metadata store. Returns a Map from the
-   * pending segment ID to the sequence name.
+   * search interval and has a sequence_name that begins with one of the prefixes in sequenceNamePrefixFilter
+   * from the metadata store. Returns a Map from the pending segment ID to the sequence name.
    */
+  @VisibleForTesting
```

Comment:
Avoid using this. It is okay to just test the method which invokes this internally.

Reply:
The method invoking this internally would ignore segments that are not needed eventually, and it would be hard to test this optimization.

Reply:
Sorry, It seems like it filters segments that have already been upgraded, and not all the unnecessary segments. I can make this change